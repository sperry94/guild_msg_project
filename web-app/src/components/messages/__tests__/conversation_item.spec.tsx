import React from "react"
import renderer from "react-test-renderer"
import ConversationItem, { ConversationItemProps } from "../conversation_item"

describe("ConversationItem", () => {
  const cases: ConversationItemProps[] = [
    {
      conversationId: undefined,
      conversation: {
        id: "test",
        users: [
          {
            displayName: "bob",
            email: "bob@bob.com",
            photoUrl: "photoUrl",
            uid: "uid1",
          },
        ],
      },
    },
    {
      conversationId: "convo-id",
      conversation: {
        id: "test",
        users: [
          {
            displayName: "bill",
            email: "bill@bill.com",
            photoUrl: "photoUrl2",
            uid: "uid2",
          },
        ],
      },
    },
    {
      conversationId: "test",
      conversation: {
        id: "test",
        users: [
          {
            displayName: "bill",
            email: "bill@bill.com",
            photoUrl: "photoUrl2",
            uid: "uid2",
          },
        ],
      },
    },
  ]

  cases.forEach((c, index) => {
    it(`renders for case #${index + 1}`, () => {
      const tree = renderer.create(<ConversationItem {...c} />).toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
