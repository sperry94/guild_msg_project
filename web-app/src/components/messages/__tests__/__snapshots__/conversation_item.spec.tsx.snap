// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConversationItem renders for case #1 1`] = `
<li>
  <a
    aria-disabled={false}
    className="MuiButtonBase-root MuiListItem-root makeStyles-link-1 MuiListItem-gutters MuiListItem-divider MuiListItem-button"
    href="/messages/test"
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-2"
    >
      <img
        alt="bob"
        className="MuiAvatar-img"
        src="photoUrl"
      />
    </div>
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-colorInherit"
    >
      bob
    </p>
  </a>
</li>
`;

exports[`ConversationItem renders for case #2 1`] = `
<li>
  <a
    aria-disabled={false}
    className="MuiButtonBase-root MuiListItem-root makeStyles-link-1 MuiListItem-gutters MuiListItem-divider MuiListItem-button"
    href="/messages/test"
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-2"
    >
      <img
        alt="bill"
        className="MuiAvatar-img"
        src="photoUrl2"
      />
    </div>
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-colorInherit"
    >
      bill
    </p>
    <span
      className="MuiTouchRipple-root"
    />
  </a>
</li>
`;

exports[`ConversationItem renders for case #3 1`] = `
<li>
  <a
    aria-disabled={false}
    className="MuiButtonBase-root MuiListItem-root makeStyles-link-1 MuiListItem-gutters MuiListItem-divider MuiListItem-button Mui-selected"
    href="/messages/test"
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-2"
    >
      <img
        alt="bill"
        className="MuiAvatar-img"
        src="photoUrl2"
      />
    </div>
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-colorInherit"
    >
      bill
    </p>
  </a>
</li>
`;
